docker/amzn-linux-nbd/build-enclave-kernel.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -exo pipefail
+
+clonelinux() {
+
+  # Clone the amazon linux kernel repository and checkout
+  # the branch which is used for nitro enclaves
+  # You can find out the kernel version used by nitro-enclaves
+  # by running uname -a in a converted app. In this case, we
+  # use version 4.14.246
+  # For more details about supported nitro enclave kernel
+  # versions, refer to the aws-nitro-enclaves-cli github repository.
+
+  if [ -d "linux" ]; then
+    cd linux
+    git fetch
+  else
+    git clone https://github.com/amazonlinux/linux.git
+  fi
+
+  git checkout microvm-kernel-4.14.246-198.474.amzn2
+
+}
+
+buildkernel() {
+
+  # Copy the enclave kernel config file into the kernel repository
+  # The original file is available here:
+  # https://github.com/aws/aws-nitro-enclaves-cli/blob/main/blobs/x86_64/bzImage.config
+  # The config file available in this directory has been updated to
+  # support salmiac features.
+  cp ../bzImage.config .config
+
+  # Build the enclave kernel
+  make prepare && make modules_prepare && make modules && make && make bzImage
```

Comment:
there are almost 4 commands here and all failure in one of them will 
1- Make it difficult to figure out which one has failed
2- The user might copy the entire command and re run it which will make all the sub commands to re run. 
Can we separate them?

Reply:
Can we add `-j <get_number_of_cpus>` flag here as well so that make uses multiple jobs? Or has that already been set somewhere?

Reply:
I think the kernel build steps should happen sequentially. Why do you want to add the `-j` option?

Reply:
If there is something in the Makefile that relies on it being built sequentially, then we can't add `-j`. You may have better insight there. Generally, `Makefile`s are smart enough to parallelize non-dependent jobs only. But yea, that isn't always the case.
If it was parallelizable, the time taken to build would reduce a lot.

Reply:
The plain `make` and `make modules` should have `-j` added, since those are the steps that take the longest to run. It won't help as much for `make prepare`, `make modules_prepare` or `make bzImage`.